Weaver/RealmWeaver.Fody/ImportedReferences.cs

```diff
@@ -0,0 +1,353 @@
+                var T = new GenericParameter(RealmObject_GetBacklinks) { Constraints = { RealmObject } };
+                (RealmObject_GetBacklinks.ReturnType as GenericInstanceType).GenericArguments.Add(T);
+                RealmObject_GetBacklinks.GenericParameters.Add(T);
+                RealmObject_GetBacklinks.Parameters.Add(new ParameterDefinition(Types.String));
+            }
+
+            IRealmObjectHelper = new TypeReference("Realms.Weaving", "IRealmObjectHelper", Module, realmAssembly);
+
+            PreserveAttribute = new TypeReference("Realms", "PreserveAttribute", Module, realmAssembly);
+            PreserveAttribute_Constructor = new MethodReference(".ctor", Types.Void, PreserveAttribute) { HasThis = true };
+            PreserveAttribute_ConstructorWithParams = new MethodReference(".ctor", Types.Void, PreserveAttribute)
+            {
+                HasThis = true,
+                Parameters = { new ParameterDefinition(Types.Boolean), new ParameterDefinition(Types.Boolean) }
+            };
+
+            WovenAttribute = new TypeReference("Realms", "WovenAttribute", Module, realmAssembly);
+            WovenAttribute_Constructor = new MethodReference(".ctor", Types.Void, WovenAttribute)
+            {
+                HasThis = true,
+                Parameters = { new ParameterDefinition(System_Type) }
+            };
+
+            WovenPropertyAttribute = new TypeReference("Realms", "WovenPropertyAttribute", Module, realmAssembly);
+            WovenPropertyAttribute_Constructor = new MethodReference(".ctor", Types.Void, WovenPropertyAttribute) { HasThis = true };
+        }
+
+        private void InitializePropertyChanged_Fody(AssemblyNameReference propertyChangedAssembly)
+        {
+            PropertyChanged_DoNotNotifyAttribute = new TypeReference("PropertyChanged", "DoNotNotifyAttribute", Module, propertyChangedAssembly);
+            PropertyChanged_DoNotNotifyAttribute_Constructor = new MethodReference(".ctor", Types.Void, PropertyChanged_DoNotNotifyAttribute) { HasThis = true };
+        }
+
+        protected AssemblyNameReference GetOrAddFrameworkReference(string assemblyName)
+        {
+            var assembly = Module.AssemblyReferences.SingleOrDefault(a => a.Name == assemblyName);
+            if (assembly == null)
+            {
+                var corlib = (AssemblyNameReference)Types.CoreLibrary;
+                assembly = new AssemblyNameReference(assemblyName, corlib.Version)
+                {
+                    Attributes = corlib.Attributes,
+                    Culture = corlib.Culture,
+                    PublicKeyToken = corlib.PublicKeyToken
+                };
+
+                Module.AssemblyReferences.Add(assembly);
+            }
+
+            return assembly;
+        }
+
+        private sealed class NETFramework : ImportedReferences
```

Comment:
I would redesign the inheritance structure here. Instead of making the properties abstract and forcing people to initialize them in inheriting class's ctor, I would expose an abstract `GetXXAssembly` method and initialize them in the base class. This way, we would be able to inline `InitializeFrameworkMethods` and `InitializeRealm` in the base class ctor and make all properties `readonly` :D This is definitely not a blocker though, just throwing it out there :)